src/NzbDrone.Core/Indexers/Definitions/Redacted.cs

```diff
@@ -41,13 +74,286 @@ protected override IndexerCapabilities SetCapabilities()
+                        Seeders = int.Parse(result.Seeders),
+                        Peers = int.Parse(result.Leechers) + int.Parse(result.Seeders),
+                        PublishDate = DateTimeOffset.FromUnixTimeSeconds(result.GroupTime).UtcDateTime,
+                        Freeleech = result.IsFreeLeech || result.IsPersonalFreeLeech,
+                        Files = result.FileCount,
+                        Grabs = result.Snatches,
+                    };
+
+                    var category = result.Category;
+                    if (category == null || category.Contains("Select Category"))
+                    {
+                        release.Categories = _categories.MapTrackerCatToNewznab("1");
+                    }
+                    else
+                    {
+                        release.Categories = _categories.MapTrackerCatDescToNewznab(category);
+                    }
+
+                    torrentInfos.Add(release);
+                }
+            }
+
+            // order by date
+            return
+                torrentInfos
+                    .OrderByDescending(o => o.PublishDate)
+                    .ToArray();
+        }
+
+        private string GetDownloadUrl(int torrentId, bool canUseToken)
+        {
+            // AuthKey is required but not checked, just pass in a dummy variable
+            // to avoid having to track authkey, which is randomly cycled
+            var url = new HttpUri(_baseUrl)
+                .CombinePath("/torrents.php")
+                .AddQueryParam("action", "download")
+                .AddQueryParam("id", torrentId)
+                .AddQueryParam("authkey", "prowlarr")
+                .AddQueryParam("torrent_pass", _settings.Passkey)
+                .AddQueryParam("usetoken", (_settings.UseFreeleechToken && canUseToken) ? 1 : 0);
+
+            return url.FullUri;
+        }
+
+        private string GetInfoUrl(string groupId, int torrentId)
+        {
+            var url = new HttpUri(_baseUrl)
+                .CombinePath("/torrents.php")
+                .AddQueryParam("id", groupId)
+                .AddQueryParam("torrentid", torrentId);
+
+            return url.FullUri;
+        }
+    }
+
+    public class RedactedSettingsValidator : AbstractValidator<RedactedSettings>
+    {
+        public RedactedSettingsValidator()
+        {
+            RuleFor(c => c.Apikey).NotEmpty();
+        }
+    }
+
+    public class RedactedSettings : IProviderConfig
+    {
+        private static readonly RedactedSettingsValidator Validator = new RedactedSettingsValidator();
+
+        public RedactedSettings()
+        {
+            Apikey = "";
+            Passkey = "";
+            UseFreeleechToken = false;
+        }
+
+        [FieldDefinition(1, Label = "API Key", HelpText = "Redacted API Key")]
+        public string Apikey { get; set; }
+
+        [FieldDefinition(1, Hidden = HiddenType.Hidden)]
+        public string Passkey { get; set; }
```

Comment:
We shouldn't need a field definition for this, it's never seen or used as a field.